pydantic/types.py

```diff
@@ -2344,9 +2344,11 @@ def __get_pydantic_json_schema__(
         return field_schema
 
     def __get_pydantic_core_schema__(self, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        base_schema: Any = handler(source)
+        del base_schema['type']
         return core_schema.with_info_after_validator_function(
             function=self.decode,
-            schema=core_schema.bytes_schema(),
+            schema=core_schema.bytes_schema(**base_schema),
```

Comment:
Wouldn't `_convert_schema` be more appropriate here, instead of unpacking and recreating, perform the union?